my-dsl.md

```diff
@@ -1,39 +1,54 @@
 
 # Domain
 
 _Describe the language's domain in five words._
 
+External DSL applying pattern recognition 
+
 # Computational model
 
 _We don't yet have a great definition of the term "computational model".
 For now, try to come up with the clearest, most concise explanation of
 what happens when a program in your DSL runs._
 
+Regular Langages analyze and find patterns in strings of code through recursive pattern analysis where the pattern is defined by the user. For instance lets say the user wanted to find a phone number from a list of peoples personal information that may include a bunch of other information that is not impornta to the user. The user could define a regular expression to find phone numbers. This DSL would start by recusivly running through the list and it would itentify anything in the list that matches the pattern of a phone number (+1(123) 456-6789).
+
 # DSL-ness
 
 _Fowler writes about a spectrum of languages, from general-purpose languages to
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum,
 and why?_
 
+Regular Languages fall in the spectrum of more "purely" domain-specific becuase it's language is used as a tool within general-purpose languages such as java and python. Regualr languages abide by all of Fowler's definitions of a DSL and fall into the external DSL catagory. 
+
 # Internal or external?
 
 _Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+Regular Languages are an external DSL becuase it is utalized as a tool within other main stream languages such as java and python. Additionally, Regular languages are in themselves a differnt language from the host language in which they are used and require differnt syntax then the host language. 
+
 # Host language
 
 _What language(s) was (were) used to implement the DSL?_
 
+The first implamentaion of regualr expressions was created by Ken Thompson, who implemented the idea of regular expressions inside the text editor called ed. Later in the late 80's, Larry Wall built a Perl programming language that helped regular expressions become mainstream. Now, regualr expressions are supported by modern programming languages, text processing programs and text editors. They can often be found in many of todays Java and Python projects. [Link to History of Regex](https://scantopdf.com/blog/the-history-of-regex/)
+
 # Benefits
 
 _Identify one potential benefit of the DSL: how is a programmer's life made
 easier by the existence of this language?_
 
+Regular expressions make a programmer's life easier by analyzing strings for patterns. AKA pattern recogintion. This makes the programers life easier becuase they do not have to define a program to find a pattern in a string instead they can simply use a Regular expression which will find the pattern in a stringg for them.
```

Comment:
Being able to do pattern recognition for strings is really cool, and probably very helpful.